src/lib.rs

```diff
@@ -23,12 +22,18 @@ where
 
 #[derive(Debug)]
 pub struct Machine<C, E> {
+    id: u64,
     addr: Ipv4Addr,
+    ctrl: mpsc::Sender<IfaceCtrl>,
     tx: mpsc::UnboundedSender<C>,
     rx: mpsc::UnboundedReceiver<E>,
 }
 
 impl<C: Send + 'static, E: Send + 'static> Machine<C, E> {
+    pub fn id(&self) -> u64 {
+        self.id
+    }
+
     pub fn addr(&self) -> Ipv4Addr {
         self.addr
     }
```

Comment:
I wonder whether changing `addr` during runtime would be an interesting feature? This would require changes to the `Router` as well, right?

Reply:
that would be interesting yes